srcalc/src/test/java/gov/va/med/srcalc/domain/model/RuleTest.java

```diff
@@ -84,12 +84,42 @@ public final void testWeightLoss() throws Exception
                 3.0f,
                 rule.apply(new Rule.EvaluationContext(3.0f, values)),
                 0.0);
-
         values.put(currWeight, currWeight.makeValue(140));
         assertEquals(
                 0.0f,
                 rule.apply(new Rule.EvaluationContext(3.0f, values)),
                 0.0f);
+        values.clear();
+        values.put(currWeight, currWeight.makeValue(140));
+        assertEquals(
+        		0.0f,
+                rule.apply(new Rule.EvaluationContext(3.0f, values)),
+                0.0f);
+    }
+    
+    @Test(expected=MissingValueException.class)
+    public final void testRequired() throws Exception
```

Comment:
It looks like this test tests both required=false and required=true. We should split this into two tests to avoid false positives (if the first call to `apply()` throws `MissingValueException`).